File: apps/agor-android/app/src/main/kotlin/live/agor/app/models/Repo.kt
package live.agor.app.models

import androidx.compose.runtime.Immutable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Immutable
@Serializable
data class Repo(
    @SerialName("repo_id") val repoId: String,
    val name: String,
    val url: String? = null,
    @SerialName("default_branch") val defaultBranch: String? = null,
    val path: String? = null,
)

File: apps/agor-rust-android/src/models/repo.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct Repo {
    pub repo_id: String,
    pub name: String,
    pub url: Option<String>,
    pub default_branch: Option<String>,
    pub path: Option<String>,
}
